src/Reveal.Sdk.Dom.Tests/Visualizations/VisualizationFixture.cs

```diff
@@ -0,0 +1,445 @@
+        }
+
+        [Fact]
+        public void FilterBindings_EqualDataDefinitionBindings_WhenUsed()
+        {
+            // Arrange
+            var dataSource = new DataSource();
+            var dataSourceItem = new DataSourceItem
+            {
+                DataSource = dataSource,
+                HasTabularData = true,
+                Fields = new List<IField>
+                {
+                    new TextField("A"),
+                    new TextField("B"),
+                    new TextField("C"),
+                    new TextField("D"),
+                }
+            };
+            var mock = new Mock<Visualization>("testTitle", dataSourceItem) { CallBase = true };
+            var visualization = mock.Object;
+
+            // Act
+            var filterBindings = visualization.FilterBindings;
+
+            // Assert
+            Assert.Equal(((DataDefinitionBase)visualization.DataDefinition).Bindings.Bindings, filterBindings);
+        }
+
+        [Fact]
+        public void GetDataSourceItem_ReturnsCorrectDataSourceItem_WhenUseTabularData()
+        {
+            // Arrange
+            var dataSource = new DataSource()
+            {
+                Id = "DataSourceId",
+            };
+            var dataSourceResourceItem = new DataSource()
+            {
+                Id = "dataSourceResourceItemId",
+            };
+            var dataSourceItem = new DataSourceItem
+            {
+                DataSource = dataSource,
+                HasTabularData = true,
+                Fields = new List<IField>
+                {
+                    new TextField("A"),
+                    new TextField("B"),
+                    new TextField("C"),
+                    new TextField("D"),
+                }
+            };
+            dataSourceItem.ResourceItem = new DataSourceItem()
+            {
+                DataSource = dataSourceResourceItem,
+                Id = "ResourceItem",
+            };
+
+            var mock = new Mock<Visualization>("testTitle", dataSourceItem) { CallBase = true };
+            var visualization = mock.Object;
+
+            var document = new RdashDocument();
+            document.DataSources.Add(dataSourceResourceItem);
+            document.Visualizations.Add(visualization);
+
+            RdashDocumentValidator.Validate(document);
+
+            // Act
+            var dataSourceItemResult = visualization.GetDataSourceItem();
+
+            // Assert
+            Assert.Equal(dataSourceItem, dataSourceItemResult);
+            Assert.Equal((visualization.DataDefinition as TabularDataDefinition).Fields.Count, dataSourceItemResult.Fields.Count);
+            Assert.Equal(dataSourceItemResult.DataSource.Id, dataSource.Id);
+            Assert.Equal(dataSourceItemResult.ResourceItemDataSource.Id, dataSourceResourceItem.Id);
+        }
+
+        [Fact]
+        public void UpdateDataSourceItem_UpdateNewDataSourceItem_AsProvided()
```

Comment:
We need to cover the case that the DataDefinition is (is not) TabularDataDefinition type